### PR TITLE
NodeGraph: Allow to set icon size and font size in dataframe 

### LINKED ISSUE #79091: NodeGraph: Allow to set icon size and font size in dataframe
**Why is this needed**:

Currently admin is unable to set different icon sizes and font sizes for nodes in a graph.

**What would you like to be added**:

Allow to have optional iconSize and fontSize columns in nodes dataframe to control these features.

**Who is this feature for?**

Admin users managing node graph.

### PR DESCRIPTION
**Pull request by piggito**
**What is this feature?**

Allow to have optional iconSize and fontSize column in nodes dataframe to control node's icon size and text font size.

**Why do we need this feature?**

Currently all nodes have fixed icon size and font size and this features adds flexibility to adapt them for better visualization.

**Who is this feature for?**

Admin users managing node graph

**Which issue(s) does this PR fix?**:

Fixes #79091

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by imatwawana — APPROVED** [review]
Thanks for the contribution! I've made some suggestions to the documentation changes, but this still requires technical review.

File: public/app/plugins/panel/nodeGraph/Node.tsx [hunk 1]
```diff
@@ -86,13 +88,20 @@ export const Node = memo(function Node(props: {
   const isHovered = hovering === 'active';
   const nodeRadius = node.nodeRadius?.values[node.dataFrameRowIndex] || nodeR;
   const strokeWidth = computeNodeCircumferenceStrokeWidth(nodeRadius);
+  const fontSize = (node.fontSize?.values[node.dataFrameRowIndex] || fontS) + 'px';
```
Comment (mckn) [inline public/app/plugins/panel/nodeGraph/Node.tsx:91]: Should we do any runtime validation with a fallback that the dataFrame contains a numeric and valid fontSize?

File: public/app/plugins/panel/nodeGraph/Node.tsx [hunk 2]
```diff
@@ -152,10 +161,27 @@ function NodeContents({ node, hovering }: { node: NodeDatum; hovering: HoverStat
     return null;
   }
 
+  const iconSize = node.iconSize?.values[node.dataFrameRowIndex] || iconS;
```
Comment (mckn) [inline public/app/plugins/panel/nodeGraph/Node.tsx:164]: Should we do any runtime validation with a fallback that the dataFrame contains  a valid iconSize?

File: packages/grafana-ui/src/components/Icon/Icon.tsx [hunk 3]
```diff
@@ -67,6 +67,7 @@ export const Icon = React.forwardRef<SVGElement, IconProps>(
         title={title}
         className={composedClassName}
         style={style}
+        icon-size={size}
```
Comment (mckn) [inline packages/grafana-ui/src/components/Icon/Icon.tsx:70]: Is this a valid attribute to pass to the SVG component or should the icon size be converted to the correct styles?

**Review by mckn — COMMENTED** [review]
Great addition! Added a couple of questions and nits.

Do we want to control this on the data frame level or should this be configured for the entire visualisation? Both are OK but I'm just checking if there is a valid use case to have different sizes on nodes within the same visualization?

**Comment by aocenas** [comment]
@piggito what is your usecase where you would use this? I have some concerns with these options
1. I don't think icon size should be separate from node size. Is there a reason for having big node and small icon or vice versa?
2. Not sure it would make sense for the data to control font size. I think font size should be something the user should be able to change based on how well they can read the labels. Maybe it could also be tied to node size so smaller nodes labels will also get demphasized but not sure it makes sense to be a separate option.

**Comment by fabrizio-grafana** [comment]
> @piggito what is your usecase where you would use this? I have some concerns with these options
> 
> 1. I don't think icon size should be separate from node size. Is there a reason for having big node and small icon or vice versa?
> 2. Not sure it would make sense for the data to control font size. I think font size should be something the user should be able to change based on how well they can read the labels. Maybe it could also be tied to node size so smaller nodes labels will also get demphasized but not sure it makes sense to be a separate option.

Hey @piggito, if you have the chance to answer these questions, that would help us in reviewing your PR. For the moment I'll close the it, since it has been stale for more than a month, but feel free to reopen it if you happen to answer the questions! Thanks!

**Pull request closed without merging**